visualization/app/codeCharta/util/urlExtractor.ts

```diff
@@ -46,8 +46,22 @@ export class UrlExtractor {
 			const responseData: string | ExportCCFile | ExportWrappedCCFile = response.data
 			const content: ExportCCFile = getCCFileAndDecorateFileChecksum(responseData)
 
+			fileName = this.getFileName(fileName, content.projectName)
+
 			return { fileName, fileSize: response.data.toString().length, content }
 		}
 		throw new Error(`Could not load file "${fileName}"`)
 	}
+
+	getFileName(oldFileName: string, projectName: string) {
+		let fileName: string
+
+		if (projectName?.trim()) {
+			fileName = projectName
+		} else {
+			const fileNameParts = oldFileName.split("/")
+			fileName = fileNameParts[fileNameParts.length - 1]
+		}
+		return fileName
```

Comment:
I suggest to return early (in line 60) and omit the else block. This can then be optimized like so:
```suggestion
		return projectName?.trim().length >0 ? projectName : oldFileName.split("/").pop();
```

Reply:
That's nice, thanks! I didn't think of the pop method

Reply:
```js
return projectName?.trim() || oldFileName.split("/").pop();
```